clients/src/main/java/org/apache/kafka/common/network/ChannelBuilders.java

```diff
@@ -129,6 +136,22 @@ private static ChannelBuilder create(SecurityProtocol securityProtocol,
                     jaasContexts = new HashMap<>(enabledMechanisms.size());
                     for (String mechanism : enabledMechanisms)
                         jaasContexts.put(mechanism, JaasContext.loadServerContext(listenerName, mechanism, configs));
+
+                    // SSL client authentication is enabled in brokers for SASL_SSL only if listener-prefixed config is specified.
+                    if (listenerName != null && securityProtocol == SecurityProtocol.SASL_SSL) {
+                        String configuredClientAuth = (String) configs.get(BrokerSecurityConfigs.SSL_CLIENT_AUTH_CONFIG);
+                        String listenerClientAuth = (String) config.originalsWithPrefix(listenerName.configPrefix(), true)
+                                .get(BrokerSecurityConfigs.SSL_CLIENT_AUTH_CONFIG);
+                        if (listenerClientAuth == null) {
+                            sslClientAuthOverride = SslClientAuth.NONE.name().toLowerCase(Locale.ROOT);
```

Comment:
I was a bit confused by this at first as I was wondering why `listenerClientAuth` is not set to `listenerClientAuth` here are well. `null` means that we let the ssl factory get the configuration from `configs` later on. We could perhaps add a comment to make this clearer.

Alternatively, could we just override `SSL_CLIENT_AUTH_CONFIG` in `configs` directly here instead of doing it in `SslFactory#configure`? This might make the handling clearer here. I don't feel strong about this. What do you think?

Reply:
We let SslFactory deal with the override since it needs to apply the override when it is reconfigured.  Added a comment. Let me know what you think.

Reply:
That makes sense. I did not think about the reconfiguration case.